packages/thirdweb/src/wallets/smart/lib/userop.ts

```diff
@@ -113,24 +118,88 @@
   } = args;
   const chain = executeTx.chain;
   const client = executeTx.client;
-  const isDeployed = await isContractDeployed(accountContract);
-  const initCode = isDeployed
-    ? "0x"
-    : await getAccountInitCode({
-        factoryContract: factoryContract,
-        adminAddress,
-        accountSalt: overrides?.accountSalt,
-        createAccountOverride: overrides?.createAccount,
-      });
-  const callData = await encode(executeTx);
+
   const bundlerOptions = {
     client,
     chain,
     entrypointAddress: overrides?.entrypointAddress,
   };
 
+  const entrypointVersion = getEntryPointVersion(
+    args.overrides?.entrypointAddress || ENTRYPOINT_ADDRESS_v0_6,
+  );
+
+  const [isDeployed, callData, gasFees, nonce] = await Promise.all([
+    isContractDeployed(accountContract),
+    encode(executeTx),
+    getGasFees({
+      executeTx,
+      bundlerOptions,
+      chain,
+      client,
+    }),
+    getAccountNonce({
```

Comment:
What does this do if the account isn't deployed?

Reply:
this actually fetches the nonce from the Entrypoint, so its always available